### PR TITLE
Add How to extend extension

### PR DESCRIPTION
**Pull request by lulco**

### CODE REVIEW

File: docs/extension.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+- assigns to [$this->template->foo = 'bar';](../src/LatteContext/Collector/VariableCollector/AssignToTemplateVariableCollector.php)
+- calls [$this->template->setParameters([...]);](../src/LatteContext/Collector/VariableCollector/SetParametersToTemplateVariableCollector.php)
+- and more, see all [variable collectors](../src/LatteContext/Collector/VariableCollector)
+
+You can implement your own variable collector implementing [`VariableCollectorInterface`](../src/LatteContext/Collector/VariableCollector/VariableCollectorInterface.php).
+Then register it as a new service in config file.
+
+## Template path collectors
+TemplatePathCollector is a service which is used to find path to latte templates.
+Now there is only one collector which finds `$template->setTemplate($path)`. So if you use some other way how to tell where the latte template is, feel free to implement [`TemplatePathCollectorInterface`](../src/LatteContext/Collector/TemplatePathCollector/TemplatePathCollectorInterface.php). Don't forget to register this new service in config file.
```
Comment (MartinMystikJonas) [inline docs/extension.md]: It is setFile. Also there shoukd be explanation how it is different from temate renders i think

File: docs/extension.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        // ...
+        $this->doSomethingWithActualClass($this->actualClass);
+        // ...
+    }    
+}
+```
+
+### Forms
+To get list of forms registered to actual Presenter or Control, you can use `FormsNodeVisitorInterface` with `FormsNodeVisitorBehavior`.
```
Comment (MartinMystikJonas) [inline docs/extension.md]: Registered to -> available in template

**Review by MartinMystikJonas — APPROVED** [review]

File: docs/extension.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
+# How to extend this extension
+
+This extension supports only common built-in methods for resolving paths to latte files, collecting variables, components, forms etc.
+However, we know that in a wild there are applications which use their own methods for these purposes. That's why we create this extension extensible.
+
```
Comment (MartinMystikJonas) [inline docs/extension.md:5]: Add table of contents

File: docs/extension.md (hunk 1 above)
Comment (lulco) [reply docs/extension.md]: nice catch! thanks

**Pull request merged**